packages/date-picker/src/picker.vue

```diff
@@ -764,6 +764,10 @@ export default {
             this.picker[option] = options[option];
           }
         }
+
+        if (this.format) {
+          this.picker.format = this.format;
```

Comment:
I suggest adding a comment here noting this is to prevent user from setting undocumented `picker.format` (Like the one you added in the test).

I might be puzzling at first because there is a `$watch('format')` before.

Reply:
The rest of PR LGTM

Reply:
@wacky6, good think, this some unobvious place. I added comment for this line.